main.py

```diff
@@ -1,280 +1,299 @@
-      Cipher_file = open(rReplace + "\\" + Username + "_decode_data.txt", "w")
-      Cipher_file.close()
-
-    if os.path.getsize(rReplace + "\\" + Username + "_decode_data.txt") > 0:
-        with open(rReplace + "\\" + Username + "_decode_data.txt", "r") as f:
-            data = f.read()
-        user_cipher_data_from_file = json.loads(data)
-        user_cipher_data_from_file = json.loads(user_cipher_data_from_file)
-    else:
-        user_cipher_data_from_file = {}
-
-    Pass = 0
-    while Pass == 0:
-        print("Please enter the name of the person you would like to decode a message from...")
-        recipient = input().lower()
-        if recipient in user_cipher_data_from_file.keys():
-            Seed = user_cipher_data_from_file[recipient]
-            Pass = 1
-
-        else:
-            print(recipient + " is not in your saved data. Would you like to create a new key for " + recipient +"? (Yes or No)")
-            Check = input().lower()
-            if Check in Check_yes or Check == "":
-                print("Please enter " + recipient + "'s ID code.")
-                Pass = 0
-                while Pass == 0:
-                  Seed = input()
-                  if Seed == "":
-                    print("Please entera vaild ID for" + recipient)
-                  else:
-                     Pass = 1
-                user_cipher_data_from_file[recipient] = Seed
-
-                user_cipher_data_from_file_json_format = json.dumps(user_cipher_data_from_file)
-                with open(rReplace + "\\" + Username + "_decode_data.txt", "w") as file:
-                  json.dump(user_cipher_data_from_file_json_format, file)
-                Pass = 1
+                print(
+                    "Are you sure you want your username to be "
+                    + new_username
+                    + "? (Yes or No)"
+                )
+                check_username = input().lower()
+                if check_username in CHECK_YES or check_username == "":
+                    print("Your username is " + new_username)
+                    PASS_ = 1
+                else:
+                    print("Please enter a username...")
+                    new_username = input()
+                    PASS_ = 0
+        PASS_ = 0
+
+        while PASS_ == 0:
+            print("Please enter a password...")
+            new_password = input()
+            print("Your pasword is " + new_password)
+            print(
+                "Are you sure you want you password to be "
+                + new_password
+                + "? (Yes or No)"
+            )
+            check_password = input().lower()
+            if check_password in CHECK_YES or check_password == "":
+                PASS_ = 1
+                USER_DATA[new_username] = new_password
+                USER_DATA_double_quotes = json.dumps(USER_DATA)
+
+                with open(new_username + ".txt", "w+", encoding="utf8") as file:
```

Comment:
Every `open()` should have an `encoding="utf8"` … I probably should have just ignored this lint error as I'm not going to explain it right now. 😬